.github/workflows/cuda-gpu-ci.yml

```diff
@@ -0,0 +1,63 @@
+name: CUDA GPU
+on:
+  workflow_dispatch:
+    inputs:
+      pr_id:
+        description: Test the contents of this Pull Request
+        required: true
+
+permissions: read-all
+
+jobs:
+  tests:
+    runs-on: GPU
```

Comment:
[Based on the doc](
https://docs.github.com/en/actions/using-github-hosted-runners/about-larger-runners/running-jobs-on-larger-runners#using-labels-and-groups-to-control-where-jobs-are-run), it seems we should be explicit about how we select the runner, e.g. either by label, by group name or both:

```suggestion
    runs-on:
        group: GPU
        labels: GPU
```

Note that both our runner label and our runner group are named "GPU". Maybe the group could be renamed to `cuda-gpu-runner-group` to be more explicit (and the label "cuda-gpu" ;)

Reply:
I renamed the group and the runner.

What made you think that this explains why things were hanging for Thomas?

Reply:
Looking at it again, I think the hanging issue will still remain, but I think it's a mostly a non-issue. I do not think many will try to open pull requests and try to invoke the GPU runner.